src/cascade/core/cascade_plan.py

```diff
@@ -0,0 +1,60 @@
+"""
+Specification for a whole cascade.
+"""
+import networkx as nx
+
+from cascade.input_data.db.locations import location_hierarchy, location_id_from_location_and_level
+from cascade.core import getLoggers
+CODELOG, MATHLOG = getLoggers(__name__)
+
+
+class CascadePlan:
+    """
+    Clients are the EpiViz-AT runner and an interactive user for configuration.
+    Collaborators are tools to build a DismodFile object and process its output.
+    Responsible for strategies
+
+     * to build a model, and
+     * to bootstrap the next model.
+
+    This knows the hierarchy of Dismod-AT models that must be fit,
+    simulated, predicted, and aggregated. Each model in the hierarchy
+    has a unique identifier of the form ``(location_id, index)`` where
+    the index would be 0 for the initial fit and increment for each
+    simulation, for instance.
+
+    Each location is at a level in the hierarchy. Most of the specification
+    depends on the level of the location within the hierarchy.
+    """
+
+    def __init__(self):
+        """
+        There are two kinds of clients, EpiViz-AT and interactive users.
+        EpiViz-AT uses its settings to parameterize this class. Both
+        clients rely on default policies.
+        """
+        self.locations = None
+        self.task_graph = None
+
+    @property
+    def tasks(self):
+        return nx.lexicographical_topological_sort(self.task_graph)
+
+    @classmethod
+    def from_epiviz_configuration(cls, execution_context, settings):
+        plan = cls()
+        plan.policies = settings.policies
+
+        plan.locations = location_hierarchy(execution_context)
+        starting_level = settings.model.split_sex
+        end_location = settings.model.drill_location
+        drill = location_id_from_location_and_level(execution_context, end_location, starting_level)
+        MATHLOG.info(f"drill nodes {', '.join(str(d) for d in drill)}")
+        tasks = [(drill_location, 0) for drill_location in drill]
```

Comment:
It doesn't get used in this drill, so maybe that was early. I wanted not to rely on a keyspace that's exactly the locations because, for instance, we want to run "global fixed" then "global random", which I would code with this method as (1, 0) and (1, 1). Or we could code them as (1, "preglobal") and (1, "fullglobal") or something. Too early for this feature.

Reply:
Ok. That's a reasonable design though.